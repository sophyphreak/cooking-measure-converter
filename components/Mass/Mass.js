import React from 'react'
import TableRow from '../../containers/TableRow/TableRow'
import getPropsForTableRow from './getPropsForTableRow/getPropsForTableRow'

const Mass = ({ massProps }) => {
  const {
    metricDropdown,
    metricInput,
    imperialInput,
    imperialDropdown
  } = getPropsForTableRow(massProps)
  return (
    <TableRow
      metricDropdown={metricDropdown}
      metricInput={metricInput}
      imperialInput={imperialInput}
      imperialDropdown={imperialDropdown}
    />
  )
}

export default Mass
