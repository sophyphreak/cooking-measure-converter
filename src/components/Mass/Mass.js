import React from 'react';
import TableRow from '../TableRow/TableRow';

const Mass = ({ massProps }) => {
  const {
    mass,
    onImperialMassChange,
    onMetricMassChange,
    massUnitOptions
  } = massProps;
  const { imperialMass, imperialUnit, metricMass, metricUnit } = mass;
  const { imperialUnitOptions, metricUnitOptions } = massUnitOptions;
  const metricDropdown = {
    value: metricUnit,
    onChange: onMetricMassChange,
    options: metricUnitOptions
  };
  const metricInput = {
    value: metricMass,
    onChange: onMetricMassChange
  };
  const imperialInput = {
    value: imperialMass,
    onChange: onImperialMassChange
  };
  const imperialDropdown = {
    value: imperialUnit,
    onChange: onImperialMassChange,
    options: imperialUnitOptions
  };
  return (
    <TableRow
      metricDropdown={metricDropdown}
      metricInput={metricInput}
      imperialInput={imperialInput}
      imperialDropdown={imperialDropdown}
    />
  );
};

export default Mass;
