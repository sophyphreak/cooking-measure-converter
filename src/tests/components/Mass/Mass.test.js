import React from 'react';
import { shallow } from 'enzyme';
import Mass from '../../../components/Mass/Mass';
import massUnitOptions from '../../../containers/MeasureConverter/unitOptions/massOptions';

let massState, onImperialMassChange, onMetricMassChange, massProps, wrapper;

beforeEach(() => {
  massState = {
    imperialMass: '',
    imperialUnit: 'lb',
    metricMass: '',
    metricUnit: 'kg'
  };
  onImperialMassChange = jest.fn();
  onMetricMassChange = jest.fn();
  massProps = {
    massState,
    onImperialMassChange,
    onMetricMassChange,
    massUnitOptions
  };
  wrapper = shallow(<Mass massProps={massProps} />);
});

test('should render Mass correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
