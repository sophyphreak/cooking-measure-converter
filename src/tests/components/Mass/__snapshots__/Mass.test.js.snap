// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Mass correctly 1`] = `
<TableRow
  imperialDropdown={
    Object {
      "onChange": [MockFunction],
      "options": Array [
        "lb",
        "oz",
      ],
      "value": "lb",
    }
  }
  imperialInput={
    Object {
      "onChange": [MockFunction],
      "value": "",
    }
  }
  metricDropdown={
    Object {
      "onChange": [MockFunction],
      "options": Array [
        "kg",
        "g",
      ],
      "value": "kg",
    }
  }
  metricInput={
    Object {
      "onChange": [MockFunction],
      "value": "",
    }
  }
/>
`;
